lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -9,185 +9,139 @@ class AwsCompileFunctions {
-      'deploy:compileFunctions': this.compileFunctions.bind(this),
+      'deploy:compileFunctions': this.compileFunctions,
     };
   }
 
-  compileFunctions() {
-    if (typeof this.serverless.service.provider.iamRoleARN !== 'string') {
-      // merge in the iamRoleLambdaTemplate
-      const iamRoleLambdaExecutionTemplate = this.serverless.utils.readFileSync(
-        path.join(this.serverless.config.serverlessPath,
-          'plugins',
-          'aws',
-          'deploy',
-          'compile',
-          'functions',
-          'iam-role-lambda-execution-template.json')
-      );
-
-      _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
-        iamRoleLambdaExecutionTemplate);
-
-      // merge in the iamPolicyLambdaTemplate
-      const iamPolicyLambdaExecutionTemplate = this.serverless.utils.readFileSync(
-        path.join(this.serverless.config.serverlessPath,
-          'plugins',
-          'aws',
-          'deploy',
-          'compile',
-          'functions',
-          'iam-policy-lambda-execution-template.json')
-      );
-
-      _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
-        iamPolicyLambdaExecutionTemplate);
-
-      // set the necessary variables for the IamPolicyLambda
-      this.serverless.service.provider.compiledCloudFormationTemplate
-        .Resources
-        .IamPolicyLambdaExecution
-        .Properties
-        .PolicyName = `${this.options.stage}-${this.serverless.service.service}-lambda`;
-      this.serverless.service.provider.compiledCloudFormationTemplate
-        .Resources
-        .IamPolicyLambdaExecution
-        .Properties
-        .PolicyDocument
-        .Statement[0]
-        .Resource = `arn:aws:logs:${this.options.region}:*:*`;
-
-      // add custom iam role statements
-      if (this.serverless.service.provider.iamRoleStatements &&
-        this.serverless.service.provider.iamRoleStatements instanceof Array) {
-        this.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources
-          .IamPolicyLambdaExecution
-          .Properties
-          .PolicyDocument
-          .Statement = this.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources
-          .IamPolicyLambdaExecution
-          .Properties
-          .PolicyDocument
-          .Statement.concat(this.serverless.service.provider.iamRoleStatements);
-      }
+  compileFunction(functionName) {
+    const newFunction = this.cfLambdaFunctionTemplate();
+    const functionObject = this.serverless.service.getFunction(functionName);
+
+    const artifactFilePath = this.serverless.service.package.individually ?
+      functionObject.artifact :
+      this.serverless.service.package.artifact;
+
+    if (!artifactFilePath) {
+      throw new Error(`No artifact path is set for function: ${functionName}`);
+    }
+
+    if (this.bucketName) {
+      newFunction.Properties.Code.S3Bucket = this.bucketName;
```

Comment:
Also, I had forgotten to run the deployment locally after the last changes.  I ran them again now and everything looks as I was expecting it.